pkg/collector/osmlogs_collector.go

```diff
@@ -123,7 +132,61 @@ func collectPodConfigs(collector *OsmLogsCollector, rootPath, meshName, namespac
 	return nil
 }
 
-// **collectControllerLogs collects logs of every OSM controller in a given mesh **
+// ** Collects Envoy proxy config for pods in monitored namespace: port-forward and curl config dump **
+func collectProxyConfigs(collector *OsmLogsCollector, rootPath, meshName, namespace string) error {
+	pods, err := utils.GetResourceList([]string{"get", "pods", "-n", namespace, "-o", "jsonpath={..metadata.name}"}, " ")
+	if err != nil {
+		return err
+	}
+	for _, podName := range pods {
+		pid, err := utils.RunBackgroundCommand("kubectl", "port-forward", podName, "-n", namespace, "15000:15000")
+		if err != nil {
+			log.Printf("Failed to collect Envoy config for pod %s in OSM monitored namespace %s: %+v", podName, namespace, err)
+			continue
+		}
+
+		var (
+			resp    *http.Response
+			retries int = 5
+		)
+		for retries > 0 {
+			resp, err = http.Get("http://localhost:15000/config_dump")
+			if err != nil {
+				retries -= 1
+				log.Printf("Error curling config dump: %+v. %d retries remaining...\n", err, retries)
+				time.Sleep(5 * time.Second)
+			} else {
+				break
+			}
+		}
+
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
```

Comment:
not sure if this handles what happens if all the retries failed. Would the resp be empty? I.e., would an `if resp != nil` block around the next section deal with that?